x-pack/winlogbeat/module/sysmon/config/winlogbeat-sysmon.js

```diff
@@ -361,7 +372,44 @@ var sysmon = (function () {
         evt.Delete("winlog.event_data.DestinationIsIpv6");
     };
 
-    var addHashes = function (evt, hashField) {
+    var setRelatedIP = function (evt) {
+        var sourceIP = evt.Get("source.ip");
+        if (sourceIP) {
+            evt.AppendTo("related.ip", sourceIP);
+        }
+
+        var destIP = evt.Get("destination.ip");
+        if (destIP) {
+            evt.AppendTo("related.ip", destIP);
+        }
+    };
+
+    var getHashPath = function (namespace, hashKey) {
+        if (hashKey === "imphash") {
+            return namespace + ".pe.imphash";
+        }
+
+        return namespace + ".hash." + hashKey;
+    };
+
+    var emptyHashRegex = new RegExp(/^0*$/);
+
+    var hashIsEmpty = function (value) {
+        if (!value) {
+            return true;
+        }
+
+        var match = emptyHashRegex.exec(value);
```

Comment:
Can you use `.test` instead of `.exec` since you don't need matches. This is usually more efficient.